packages/ckeditor5-engine/src/model/batch.js

```diff
@@ -37,17 +62,32 @@ export default class Batch {
 		this.operations = [];
 
 		/**
-		 * The type of the batch.
+		 * Whether batch can be undone through the undo feature.
 		 *
-		 * It can be one of the following values:
-		 * * `'default'` &ndash; All "normal" batches. This is the most commonly used type.
-		 * * `'transparent'` &ndash; A batch that should be ignored by other features, i.e. an initial batch or collaborative editing
-		 * changes.
+		 * @type {Boolean}
+		 */
+		this.isUndoable = isUndoable;
+
+		/**
+		 * Whether batch includes operations created locally (`true`) or operations created on other, remote editors (`false`).
 		 *
-		 * @readonly
-		 * @type {'transparent'|'default'}
+		 * @type {Boolean}
+		 */
+		this.isLocal = isLocal;
+
+		/**
+		 * Whether batch was created by the undo feature and undoes other operations.
+		 *
+		 * @type {Boolean}
+		 */
+		this.isUndo = isUndo;
+
+		/**
+		 * Whether batch includes operations connected with typing.
+		 *
+		 * @type {Boolean}
 		 */
-		this.type = type;
```

Comment:
Not sure if this is needed. We will already add breaking changes entry about that (just added it). I would be fine without it. If we add this, we should also add an issue to remove it after some time.

So, it would be something like:

```javascript
get type() {
	logWarning( ... );

	return 'default';
}
```

?